### PR TITLE
docs(AzureRedisInstance): user documentation

### LINKED ISSUE #448: [Docs] AzureRedisInstance docs
**Description**

AzureRedisInstance user docs.

**Area**

**Reasons**

**Assignees**

@kyma-project/technical-writers

**Attachments**

### PR DESCRIPTION
**Pull request by ijovovic**
**Description**

Changes proposed in this pull request:

- Description of the AzureRedisInstance CR
- Spec parameters description
- Example yaml file

**Related issue(s)**
Resolves #448

### CODE REVIEW

File: docs/user/resources/04-40-10-azure-redis-instance.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# AzureRedisInstance Custom Resource
+
+The `azureredisinstance.cloud-resources.kyma-project.io` custom resource (CR) defines an
+[Azure Cache for Redis instance](https://azure.microsoft.com/en-us/products/cache) that can be used as a caching solution within the cluster. 
+Please note that the current implementation does not support backups, so using `AzureRedisInstance` may not be ideal 
+for permanent storage. The endpoint and secret token for the instance are provided
+in a Secret resource, which is generated within the Kyma cluster. This Secret will share the same name as the `AzureRedisInstance`.
```
Comment (IwonaLanger) [inline docs/user/resources/04-40-10-azure-redis-instance.md:7]:
```suggestion
> [!NOTE] 
> The current implementation does not support backups, so using `AzureRedisInstance` is not ideal for permanent storage. 
> The endpoint and secret token for the instance are provided in a Secret resource, which is generated within the Kyma cluster. This Secret shares the same name as the `AzureRedisInstance`.
```

File: docs/user/resources/04-40-10-azure-redis-instance.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+The Azure Cache for Redis `geo-location` will be set to match the location of the Kyma cluster where the manifest is applied.
```
Comment (IwonaLanger) [inline docs/user/resources/04-40-10-azure-redis-instance.md:9]:
```suggestion
The Azure Cache for Redis `geo-location` is set to match the location of the Kyma cluster where the manifest is applied.
```

File: docs/user/resources/04-40-10-azure-redis-instance.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+The underlying Redis tier is Premium, as it supports VirtualNetwork.
+
+## Specification <!-- {docsify-ignore} -->
+This table lists the parameters of the given resource together with their descriptions. Note that this table is subject
+to change, as the underlying Azure API could change with the new versions. Therefore, use it is basic info and for any
+in-depth spec, please refer to [MS Azure latest Redis documentation](https://learn.microsoft.com/en-us/rest/api/redis/redis/create).
```
Comment (IwonaLanger) [inline docs/user/resources/04-40-10-azure-redis-instance.md:16]:
```suggestion
This table lists the parameters of the given resource together with their descriptions. The table is subject
to change, as the underlying Azure API can change with the new versions. Use it as basic information, and for
in-depth specification, see the [MS Azure latest Redis documentation](https://learn.microsoft.com/en-us/rest/api/redis/redis/create). 
```

File: docs/user/resources/04-40-10-azure-redis-instance.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+The underlying Redis tier is Premium, as it supports VirtualNetwork.
+
+## Specification <!-- {docsify-ignore} -->
+This table lists the parameters of the given resource together with their descriptions. Note that this table is subject
+to change, as the underlying Azure API could change with the new versions. Therefore, use it is basic info and for any
+in-depth spec, please refer to [MS Azure latest Redis documentation](https://learn.microsoft.com/en-us/rest/api/redis/redis/create). 
+
+Also note that not all the parameters supported on the Azure side can be used from Kyma 
+(eg `properties.redisConfiguration.aof-backup-enabled`).
```
Comment (IwonaLanger) [inline docs/user/resources/04-40-10-azure-redis-instance.md:19]:
```suggestion
> [!NOTE]
> Some parameters, such as **properties.redisConfiguration.aof-backup-enabled**, are supported by Azure, but you can't use them from Kyma.
```

File: docs/user/resources/04-40-10-azure-redis-instance.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+| **sku**                                                | object   | The SKU of the Redis cache to deploy. Required.                                                   |
+| **sku.capacity**                                       | quantity | The size of the Redis cache to deploy. Valid values: 1, 2, 3, 4. Required.                        |
+| **redisConfiguration**                                 | object   | Redis Settings. Not all settings supported on the Azure side are supported in Kyma. Optional.     |
+| **redisConfiguration.maxclients**                      | string   | The max clients config. Optional.                                                                 |
+| **redisConfiguration.maxfragmentationmemory-reserved** | string   | Value in megabytes reserved for fragmentation per shard. Optional.                                |
+| **redisConfiguration.maxmemory-delta**                 | string   | Value in megabytes reserved for non-cache usage per shard e.g. failover. Optional.                |
+| **redisConfiguration.maxmemory-policy**                | string   | The eviction strategy used when your data won't fit within its memory limit. Optional.            |
```
Comment (IwonaLanger) [inline docs/user/resources/04-40-10-azure-redis-instance.md:35]:
```suggestion
| **redisConfiguration.maxmemory-policy**                | string   | The eviction strategy used when your data doesn't fit within its memory limit. Optional.            |
```

File: docs/user/resources/04-40-10-azure-redis-instance.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+| **redisConfiguration.maxmemory-reserved**              | string   | Value in megabytes reserved for non-cache usage per shard e.g. failover. Optional.                |
```
Comment (IwonaLanger) [inline docs/user/resources/04-40-10-azure-redis-instance.md:36]:
```suggestion
| **redisConfiguration.maxmemory-reserved**              | string   | Value in megabytes reserved for non-cache usage per shard, for example, failover. Optional.                |
```

File: docs/user/resources/04-40-10-azure-redis-instance.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+| **redisConfiguration.notify-keyspace-events**          | string   | The keyspace events which should be monitored. Optional.                                          |
+| **redisConfiguration.zonal-configuration**             | string   | Zonal Configuration. Optional.                                                                    |
+
+An exemplary `AzureRedisInstance` custom resource:
```
Comment (IwonaLanger) [inline docs/user/resources/04-40-10-azure-redis-instance.md:40]:
```suggestion
See an example of the `AzureRedisInstance` custom resource:
```

File: docs/user/resources/04-40-10-azure-redis-instance.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+| **sku**                                                | object   | The SKU of the Redis cache to deploy. Required.                                                   |
+| **sku.capacity**                                       | quantity | The size of the Redis cache to deploy. Valid values: 1, 2, 3, 4. Required.                        |
+| **redisConfiguration**                                 | object   | Redis Settings. Not all settings supported on the Azure side are supported in Kyma. Optional.     |
+| **redisConfiguration.maxclients**                      | string   | The max clients config. Optional.                                                                 |
+| **redisConfiguration.maxfragmentationmemory-reserved** | string   | Value in megabytes reserved for fragmentation per shard. Optional.                                |
+| **redisConfiguration.maxmemory-delta**                 | string   | Value in megabytes reserved for non-cache usage per shard e.g. failover. Optional.                |
```
Comment (IwonaLanger) [inline docs/user/resources/04-40-10-azure-redis-instance.md:34]:
```suggestion
| **redisConfiguration.maxmemory-delta**                 | string   | Value in megabytes reserved for non-cache usage per shard, for example, failover. Optional.                |
```

File: docs/user/resources/04-40-10-azure-redis-instance.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+## Specification <!-- {docsify-ignore} -->
+This table lists the parameters of the given resource together with their descriptions. Note that this table is subject
+to change, as the underlying Azure API could change with the new versions. Therefore, use it is basic info and for any
+in-depth spec, please refer to [MS Azure latest Redis documentation](https://learn.microsoft.com/en-us/rest/api/redis/redis/create). 
+
+Also note that not all the parameters supported on the Azure side can be used from Kyma 
+(eg `properties.redisConfiguration.aof-backup-enabled`).
+
+**Spec:**
+
```
Comment (IwonaLanger) [inline docs/user/resources/04-40-10-azure-redis-instance.md:22]: Please consider adding another column for the required/optional value.

**Comment by ijovovic** [comment]
Closing the PR as new impl of Azure Redis is in and new version of docs as well: [ttps://github.com/kyma-project/cloud-manager/pull/735](https://github.com/kyma-project/cloud-manager/pull/735)

**Pull request closed without merging**